sourcetracker/_sourcetracker.py

```diff
@@ -9,195 +9,203 @@
-    """Parse lines from a mapping file into a nested dictionary.
+def check_and_correct_data(feature_table, cfc):
+    '''Check and correct problems in the `feature_table`.
 
-    This function copies qiime.parse_mapping_file_to_dict, but does _not_ allow
-    commented lines at the top of a mapping file.
+    Parameters
+    ----------
+    feature_table : pd.DataFrame
+        Contingency table with rows, columns = samples, features.
+    cfc : boolean
+        Fractional counts cause errors in indexing. Uses np.ceil to remove
+        them if True.
+
+    Returns
+    -------
+    feature_table : pd.DataFrame
+        Corrected feature table if changes have been made or original table if
+        no changes occurred. Will automatically be cast as dtype=np.int64
+
+    Raises
+    ------
+    ValueError
+        If data contains values that Gibb's sampling cannot handle.
+    '''
+    _ft = feature_table.copy()
+    if _ft.isnull().values.sum() != 0:
+        raise ValueError('One or more values in the feature table is '
+                         'non-numeric. This will cause propagating error and '
+                         'must be removed.')
+    if not (np.ceil(_ft.values) == _ft.values).all():
+        if cfc is True:
+            _ft.iloc[:, :] = np.ceil(_ft.values)
+        else:
+            raise ValueError('There are non-integer counts in the data and '
+                             'you have selected not to correct them. This '
+                             'will cause indexing errors. Data must be '
+                             'integerial.')
+    _ft.iloc[:, :] = _ft.values.astype(np.int64)
+    return _ft
+
+
+def biom_to_df(_ft, cfc=True):
+    '''Turn biom table into dataframe, correcting fractional counts.
 
     Parameters
     ----------
-    mf_lines : list
-        Each entry of the list is a tab delimited string corresponding to a row
-        of the mapping file.
+    _ft : biom.table.Table
+        Biom table.
+    cfc : boolean
+        Fractional counts cause errors in indexing. Uses np.ceil to remove
+        them if True.
 
     Returns
     -------
-    results : dict
-    """
-    headers = mf_lines[0].strip().split('\t')[1:]
-    results = {}
-    for line in mf_lines[1:]:
-        tmp = line.strip().split('\t')
-        results[tmp[0]] = {k: v.strip() for k, v in zip(headers, tmp[1:])}
-    return results
+    feature_table : pd.DataFrame
+        Contingency table with rows, columns = samples, features.
+    '''
+    ft = pd.DataFrame(_ft._data.toarray().T, index=_ft.ids(axis='sample'),
+                      columns=_ft.ids(axis='observation'))
+    return check_and_correct_data(ft, cfc=cfc)
 
 
-def collapse_sources(samples, sample_metadata, category, biom_table,
-                     sort=True):
-    """Collapse sources based on metadata under `category`.
+def remove_unshared_samples_and_order(sample_metadata, feature_table):
+    '''Return input tables with unshared samples removed, row order equivalent.
```

Comment:
I agree - would ``intersect_and_sort_samples`` be a better name for this? And then adopt that nomenclature in the documentation?